src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -51,83 +125,168 @@ public class ScriptTagHelper : TagHelper
         protected internal ILogger<ScriptTagHelper> Logger { get; set; }
 
         [Activate]
+        protected internal IHostingEnvironment HostingEnvironment { get; set; }
+
+        [Activate]
         protected internal ViewContext ViewContext { get; set; }
 
+        [Activate]
+        protected internal IMemoryCache Cache { get; set; }
+
+        // Internal for ease of use when testing.
+        protected internal GlobbingUrlBuilder GlobbingUrlBuilder { get; set; }
+
         /// <inheritdoc />
         public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
         {
-            if (!AttributeMatcher.AllRequiredAttributesArePresent(context, ViewContext, RequiredAttributes, Logger))
-            {
-                if (Logger.IsEnabled(LogLevel.Verbose))
-                {
-                    Logger.WriteVerbose(
-                        "Skipping processing for {0} with ID {1} on view '{2}'",
-                        nameof(ScriptTagHelper),
-                        context.UniqueId,
-                        ViewContext.View.Path);
-                }
+            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
+
+            Debug.Assert(modeResult.FullMatches.Select(match => match.Mode).Distinct().Count() <= 1,
```

Comment:
Is this valid: `<script asp-src-include="..." asp-fallback-src="..." asp-fallback-test="..."></script>`? If so you'll have two full match results